Automattic/wp-calypso
# Add jetpack-ai-calypso package

Related to https://github.com/Automattic/jetpack/issues/34376

## Proposed Changes

* Adds a package named `jetpack-ai-calypso` to be used for Jetpack AI features that need to be available directly on Calypso and possibly shared with Jetpack.

## Testing Instructions
- Take a look at the package configuration. Anything there that can be optimized or is wrong?
- Build the new package: `yarn` for dependencies and then `yarn start`. There should be no build errors.
- Run the JS tests with `yarn run test-packages jetpack-ai-calypso`. It is just a Hello World, intended to test if everything is rendering as intended
- Now, to test the package in use with loaded styles, add this to `client/my-sites/customer-home/cards/actions/quick-links/index.jsx`:
```jsx
3   | import { GeneratorModal } from '@automattic/jetpack-ai-calypso';
... | [...]
245 | 					<GeneratorModal />

```
There should be a "Hello World" in **bold** there:
![2023-12-12_18-15-19](https://github.com/Automattic/wp-calypso/assets/8486249/c40bd94d-199b-4a10-9e15-3c61802c850e)
- To test typescript hot reloading, run `yarn workspace @automattic/jetpack-ai-calypso run watch` and change the text. Check that it changes in the screen without a need for reloads.

This PR is focused on creating the package only. Temporary files or components will be removed in other PRs.



## Pre-merge Checklist
